bodhi/server/consumers/masher.py

```diff
@@ -1006,3 +1037,24 @@ def run(self):
         else:
             self.success = True
         return out, err, returncode
+
+def system(cmd):
+    """
+    Invoke a shell command.
+    :returns: A tuple of output, err message, and return code
+    """
+    ret = subprocess.Popen(cmd, shell=True, stdin=subprocess.PIPE,
+                stdout=subprocess.PIPE, stderr=subprocess.PIPE, close_fds=True)
+    out, err = ret.communicate()
+    return out, err, ret.returncode
+
+
+def git_checkout(dirpath, url, branch):
+    """Checks out the given url in the directory and given branch    
+    :param dirpath: Path to checkout  
+    :param url: URL to clone
+    :param branch: The branch to checkout
+    :return: None
+    """
+    cmd = "git clone -b %s %s %s" % (branch, url, dirpath)
+    system(cmd)
```

Comment:
I did think of a benefit to having a git checkout function - we could make this function gracefully handle Pagure outages by retrying. I'd recommend retrying for a relatively long time before giving up - maybe even 3-4 hours. It should log to err level on each failure so we get e-mail notifications about the problem.